src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PhysicalDetailService.kt

```diff
@@ -0,0 +1,152 @@
+package uk.gov.justice.digital.hmpps.prisonersearch.services
+
+import com.google.gson.Gson
+import com.microsoft.applicationinsights.TelemetryClient
+import org.elasticsearch.action.search.SearchRequest
+import org.elasticsearch.action.search.SearchResponse
+import org.elasticsearch.common.unit.TimeValue
+import org.elasticsearch.index.query.BoolQueryBuilder
+import org.elasticsearch.index.query.QueryBuilders
+import org.elasticsearch.search.builder.SearchSourceBuilder
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.PageImpl
+import org.springframework.data.domain.PageRequest
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.prisonersearch.model.Prisoner
+import uk.gov.justice.digital.hmpps.prisonersearch.security.AuthenticationHolder
+import uk.gov.justice.digital.hmpps.prisonersearch.services.dto.PaginationRequest
+import uk.gov.justice.digital.hmpps.prisonersearch.services.dto.PhysicalDetailRequest
+import uk.gov.justice.digital.hmpps.prisonersearch.services.exceptions.BadRequestException
+import java.util.concurrent.TimeUnit
+
+@Service
+class PhysicalDetailService(
+  private val elasticsearchClient: SearchClient,
+  private val indexStatusService: IndexStatusService,
+  private val gson: Gson,
+  private val telemetryClient: TelemetryClient,
+  private val authenticationHolder: AuthenticationHolder,
+  @Value("\${search.detailed.max-results}") private val maxSearchResults: Int = 200,
+  @Value("\${search.detailed.timeout-seconds}") private val searchTimeoutSeconds: Long = 10L,
+) {
+  companion object {
+    private val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  private fun validateDetailRequest(detailRequest: PhysicalDetailRequest): Unit = with(detailRequest) {
+    if (prisonIds.isNullOrEmpty()) {
+      throw BadRequestException("Invalid physical detail search  - please provide prison locations to filter by")
+    }
+    if ((minHeight ?: 0) < 0) {
+      throw BadRequestException("Minimum height cannot be less than 0")
+    }
+    if ((maxHeight ?: 0) < 0) {
+      throw BadRequestException("Minimum height cannot be less than 0")
```

Comment:
Typo on minimum

Reply:
took me a while here to understand your comment - I was thinking that I'd spelt minimum incorrectly 😂

Reply:
will correct on next PR if that's okay - got to add height etc.